### PR TITLE
deploy: correct updating lastTransitionTime in deployment conditions

### PR DESCRIPTION
**Pull request by kargakis**
@mfojtik @smarterclayton ptal

### CODE REVIEW

**Comment by kargakis** [comment]
[test]

File: pkg/deploy/util/util_test.go [hunk 1]
```diff
@@ -502,10 +504,21 @@ func TestSetCondition(t *testing.T) {
...
-			expectedStatus: &deployapi.DeploymentConfigStatus{Conditions: []deployapi.DeploymentCondition{condProgressingDifferentReason()}},
+			expectedStatus: &deployapi.DeploymentConfigStatus{
+				Conditions: []deployapi.DeploymentCondition{
+					{
+						Type:   deployapi.DeploymentProgressing,
+						Status: kapi.ConditionTrue,
+						// Note that LastTransitionTime stays the same.
+						LastTransitionTime: now,
+						// Only the reason changes.
+						Reason: "BecauseItIs",
```
Comment (mfojtik) [inline pkg/deploy/util/util_test.go]: lol :-)

**Comment by mfojtik** [comment]
LGTM

**Comment by smarterclayton** [comment]
LGTM [merge] thanks

**Pull request merged**